CMakeLists.txt

```diff
@@ -66,49 +78,76 @@ else()
-    target_link_libraries(${TEST_TARGET} GTest::gtest GTest::gtest_main)
-    target_include_directories(${TEST_TARGET} PRIVATE include)
-    add_test(NAME ${TEST_TARGET} COMMAND ${TEST_TARGET})
-endmacro()
-
-add_executable(comparison_chain_test test/comparison_chain_test.cpp)
-add_test_dependencies(comparison_chain_test)
-add_executable(concepts_test test/concepts_test.cpp)
-add_test_dependencies(concepts_test)
-add_executable(enum_array_test test/enum_array_test.cpp)
-add_test_dependencies(enum_array_test)
-add_executable(enum_map_test test/enum_map_test.cpp)
-add_test_dependencies(enum_map_test)
-add_executable(enum_set_test test/enum_set_test.cpp)
-add_test_dependencies(enum_set_test)
-add_executable(enum_utils_test test/enum_utils_test.cpp)
-add_test_dependencies(enum_utils_test)
-add_executable(fixed_map_test test/fixed_map_test.cpp)
-add_test_dependencies(fixed_map_test)
-add_executable(fixed_map_perf_test test/fixed_map_perf_test.cpp)
-add_test_dependencies(fixed_map_perf_test)
-add_executable(fixed_red_black_tree_test test/fixed_red_black_tree_test.cpp)
-add_test_dependencies(fixed_red_black_tree_test)
-add_executable(fixed_set_test test/fixed_set_test.cpp)
-add_test_dependencies(fixed_set_test)
-add_executable(fixed_vector_test test/fixed_vector_test.cpp)
-add_test_dependencies(fixed_vector_test)
-add_executable(index_range_predicate_iterator_test test/index_range_predicate_iterator_test.cpp)
-add_test_dependencies(index_range_predicate_iterator_test)
-add_executable(instance_counter_test test/instance_counter_test.cpp)
-add_test_dependencies(instance_counter_test)
-add_executable(pair_view_test test/pair_view_test.cpp)
-add_test_dependencies(pair_view_test)
-add_executable(string_literal_test test/string_literal_test.cpp)
-add_test_dependencies(string_literal_test)
-add_executable(type_name_test test/type_name_test.cpp)
-add_test_dependencies(type_name_test)
+add_library(project_warnings INTERFACE)
+target_compile_options(project_warnings INTERFACE ${DIAGNOSTIC_FLAGS})
+
+
+find_package(magic_enum CONFIG)
+find_package(range-v3 CONFIG)
+add_library(fixed_containers INTERFACE)
+add_library(fixed_containers::fixed_containers ALIAS fixed_containers)
+target_include_directories(fixed_containers INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
+target_link_libraries(fixed_containers INTERFACE magic_enum::magic_enum)
+target_link_libraries(fixed_containers INTERFACE range-v3)
+
+
+# TODO: split library in components like in bazel
+#add_library(fixed_containers_bidirectional_iterator INTERFACE include/fixed_containers/bidirectional_iterator.hpp)
+#add_library(fixed_containers::bidirectional_iterator ALIAS fixed_containers_bidirectional_iterator)
+#target_include_directories(fixed_containers_bidirectional_iterator INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
+
+
+option(BUILD_TESTS "Enable Tests" ${PROJECT_IS_TOP_LEVEL})
+if(BUILD_TESTS)
+    # This variable is set by project() in CMake 3.21+
```

Comment:
Per this comment, PROJECT_IS_TOP_LEVEL is only available in CMake 3.21+ and it is set here for compatibility with lower version, but the variable it used 2 lines above and controls whether this if-block is entered, so this wouldn't work for < 3.21.

The built-in `BUILD_TESTING` might be of use here: https://cmake.org/cmake/help/latest/command/enable_testing.html
According to the docs, it also calls `enable_testing()` automatically.